ravens/uml/visualize.py

```diff
@@ -75,141 +212,202 @@ def _parse_object_style(object_style_string: str):
                         object_style[key_outer] = {key_inner: value}
                     else:
                         raise ValueError
-        except ValueError as msg:
+        except ValueError:
             print(object_style_string)
 
         return object_style
 
     def _create_svg_data(self, diagram_id: int):
-        dobjects = self.uml_data.diagramobjects[self.uml_data.diagramobjects["Diagram_ID"] == diagram_id]
+        dobj = self.uml_data.diagramobjects
+        dlinks = self.uml_data.diagramlinks
+
+        dobjects = dobj[dobj["Diagram_ID"] == diagram_id]
         if dobjects.empty:
-            return {"cx": 0, "cy": 0, "nodes": [], "links": []}
+            self._current_diagram = diagram_id
+            self._current_svg_data = {"cx": 0, "cy": 0, "nodes": [], "links": []}
+            return
 
-        svg_data = {
-            "cx": max([abs(o.RectRight) for o in dobjects.itertuples()]),
-            "cy": max([abs(o.RectBottom) for o in dobjects.itertuples()]),
-        }
+        # canvas size
+        cx = int(abs(dobjects["RectRight"]).max())
+        cy = int(abs(dobjects["RectBottom"]).max())
+        svg_data = {"cx": cx, "cy": cy}
 
+        # nodes present on this diagram (set for quick membership checks)
+        node_ids = set(int(r.Object_ID) for r in dobjects.itertuples(index=False))
         boxes_data = []
-        nodes = []
-        objs_in_diagram = [_o.Object_ID for _o in self.uml_data.diagramobjects[self.uml_data.diagramobjects["Diagram_ID"] == diagram_id].itertuples()]
 
-        for o in self.uml_data.diagramobjects[self.uml_data.diagramobjects["Diagram_ID"] == diagram_id].itertuples():
-            object_style = self._parse_object_style(str(o.ObjectStyle))
-            obj = self.uml_data.objects.loc[o.Object_ID]
+        for o in dobjects.itertuples(index=False):
+            oid = int(o.Object_ID)
+            stereo = self._obj_stereo.get(oid, "")
 
+            # text lines
             text_lines = []
-            if obj.Stereotype == "enumeration":
-                text_lines.append({"text": f"<<{obj.Stereotype}>>", "align": "center"})
-                text_lines.append({"text": f"{obj.Name}", "align": "center", "style": "bold"})
+            if stereo == "enumeration":
```

Comment:
Changes here are mainly to use `.get()` instead of direct referencing for fields that are either optional or inconsistently present. Avoids Key Errors and easier to fall back to stereotype/legacy colors.